tests/protocol/test_wire.py

```diff
@@ -900,15 +1031,44 @@ def host_func(*args):
 
             # Assert we've called both the direct channel function (once) and the host channel function (twice).
             # After the host channel succeeds, the host plugin should have been set as the default channel.
-            with patch(
-                    'azurelinuxagent.common.protocol.wire.WireClient.update_host_plugin_from_goal_state') as mock_update_host_plugin_from_goal_state:
+            with patch("azurelinuxagent.common.protocol.wire.WireClient.update_host_plugin_from_goal_state") \
+                    as mock_update_host_plugin_from_goal_state:
                 ret = protocol.client.send_request_using_appropriate_channel(direct_func, host_func)
                 self.assertEquals(42, ret)
                 self.assertEquals(1, direct_func.counter)
                 self.assertEquals(2, host_func.counter)
                 self.assertEquals(1, mock_update_host_plugin_from_goal_state.call_count)
                 self.assertEquals(True, protocol.client.get_host_plugin().is_default_channel())
 
+    def test_send_request_using_appropriate_channel_should_not_set_host_channel_as_default_if_response_failed(self):
+        with mock_wire_protocol(mockwiredata.DATA_FILE) as protocol:
+            protocol.client.get_host_plugin().set_default_channel(False)
+
+            def direct_func(*args):
+                direct_func.counter += 1
+                raise InvalidContainerError()
+
+            def host_func(*args):
+                host_func.counter += 1
+                if host_func.counter == 1:
+                    raise ResourceGoneError("Resource is gone")
+                else:
+                    return None
+
+            direct_func.counter = 0
+            host_func.counter = 0
+
+            # Assert we've called both the direct channel function (once) and the host channel function (twice).
+            # The second call through host channel resulted in None, so we shouldn't set the host channel as default.
+            with patch("azurelinuxagent.common.protocol.wire.WireClient.update_host_plugin_from_goal_state") \
+                    as mock_update_host_plugin_from_goal_state:
+                ret = protocol.client.send_request_using_appropriate_channel(direct_func, host_func)
+                self.assertEquals(None, ret)
+                self.assertEquals(1, direct_func.counter)
```

Comment:
NIT: Could you add some error message incase this test fails just for improved debuggability

Reply:
Done, thanks for catching that.